virtual_rainforest/core/core_schema.json

```diff
@@ -188,10 +188,17 @@
                "type": "object",
                "properties": {
                   "soil_layers": {
-                     "description": "Number of soil layers to simulate",
-                     "type": "integer",
-                     "exclusiveMinimum": 0,
-                     "default": 2
+                     "description": "Depth and number of soil layers to simulate",
+                     "type": "array",
+                     "items": {
+                        "type": "number"
+                     },
+                     "minItems": 1,
+                     "uniqueItems": true,
+                     "default": [
+                        -0.5,
+                        -1.0
+                     ]

```

Comment:
Just a thought, but I'm wondering how much sense it really makes to have a default value at all for this param? Isn't it something v simulation specific that users will basically always want to change?

I appreciate that you've similarly got defaults for things like canopy layers already, so I'm not suggesting you remove it. I just wondered whether you had a plan to remove these defaults at some point when VR becomes more than just a skeleton.

Reply:
It's a good point and I think revising the schema defaults once we have all four models running is going to be really useful. These aren't especially simulation specific: the canopy layers is a maximum and just basically sets a maximum array size, and the soil depths are fairly standard.

One thing we may want to do with canopy layers is allow it to be dynamic because the number of actually _used_ layers varies with plant community growth. But that's a whole 'nuther PR.

Reply:
Is that info documented somewhere?

Feels like we ought to put this information into the JSONSchema or somewhere to help users populate it. I can imagine a tool _waaay_ in the future that does:

```text
> So, you want to create a VR Config?
...
> Step N: Set your soil layer depths: the top layer being where microbial activity
> happens and the second layer where soil temperature equals annual mean air 
> temperature. Edit the default below:
>>> [-0.5, -1.0]
```

Reply:
The lower level is documented in the microclimate docstring, this is where it is used to calculate soil temperature profiles (will also be in the abiotic model at some point when I implement the thomas algorithm). I'm not sure about the top soil, might be in the soil model. I agree it should be somewhere central now that we know where it is initialized

Reply:
For the moment, I've put it in the ``set_layer_roles`` docstring, so it exists.